app/src/main/java/com/github/se/signify/model/home/quest/FirestoreQuestRepository.kt

```diff
@@ -36,13 +37,15 @@ class FirestoreQuestRepository(private val db: FirebaseFirestore) : QuestReposit
       val title = document.getString("title") ?: return null
       val description = document.getString("description") ?: return null
       val index = document.getString("index") ?: return null
-
-      // This video path will change once all the videos are ready, for now all words points to the
-      // same video
-      val videoPath = "android.resource://com.github.se.signify/raw/thankyou"
+      val videoPath = fetchVideo(title)
       Quest(index = index, title = title, description = description, videoPath = videoPath)
     } catch (e: Exception) {
       null
     }
   }
+
+  internal fun fetchVideo(word: String): String {
+    val sanitizedWord = word.replace(" ", "").lowercase() // Remove spaces and convert to lowercase
+    return "$videoPathBase$sanitizedWord"
+  }
```

Comment:
The fetchVideo function lack error handling for cases where the video path is invalid or the video fails to load. 
 
**Suggestion**: Add error listeners and display a fallback UI (for example an error text or a placeholder image) when the video cannot be loaded.